File: app/src/main/java/com/angogasapps/myfamily/utils/StringFormater.java
package com.angogasapps.myfamily.utils;



import java.nio.charset.StandardCharsets;
import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.Locale;
import static com.angogasapps.myfamily.firebase.FirebaseVarsAndConsts.*;

public class StringFormater {
    public static String formatStringToSend(String text){
        StringBuilder str = new StringBuilder(text);

        while (String.valueOf(str.charAt(0)).equals(" ") || String.valueOf(str.charAt(0)).equals("\n")){
            str.deleteCharAt(0);
        }
        while (String.valueOf(str.charAt(str.length() - 1)).equals(" ") || String.valueOf(str.charAt(str.length() - 1)).equals("\n")){
            str.deleteCharAt(str.length() - 1);
        }

        return str.toString();
    }

    public static String formatLongToTime(Long time){
        SimpleDateFormat timeFormat = new SimpleDateFormat("HH:mm", Locale.getDefault());
        return timeFormat.format(new Date(time));
    }

    public static String formatToRole(String role){
        if (role == null)
            return "";
        if (role.equals(ROLE_MEMBER))
            return "Участник";
        if (role.equals(ROLE_CREATOR))
            return "Создатель";


        return role;
    }

    public static String convertUTF8ToString(String s) {
        String out = new String(s.getBytes(StandardCharsets.ISO_8859_1), StandardCharsets.UTF_8);
        return out;
    }

    public static String convertStringToUTF8(String s) {
        String out = new String(s.getBytes(StandardCharsets.UTF_8), StandardCharsets.ISO_8859_1);
        return out;
    }


}

File: app/src/main/java/com/angogasapps/myfamily/utils/StringFormater.kt
package com.angogasapps.myfamily.utils

import com.angogasapps.myfamily.app.AppApplication.Companion.getInstance
import com.angogasapps.myfamily.firebase.FirebaseVarsAndConsts
import com.angogasapps.myfamily.app.AppApplication
import com.angogasapps.myfamily.R
import java.nio.charset.StandardCharsets
import java.text.SimpleDateFormat
import java.util.*

object StringFormater {
    @JvmStatic
    fun formatLongToTime(time: Long?): String {
        val timeFormat = SimpleDateFormat("HH:mm", Locale.getDefault())
        return timeFormat.format(Date(time!!))
    }

    @JvmStatic
    fun formatToRole(role: String?): String {
        if (role == null) return ""
        if (role == FirebaseVarsAndConsts.ROLE_MEMBER) return getInstance().getString(R.string.participant)
        return if (role == FirebaseVarsAndConsts.ROLE_CREATOR) getInstance().getString(R.string.creator) else role
    }

    fun convertUTF8ToString(s: String): String {
        return String(s.toByteArray(StandardCharsets.ISO_8859_1), StandardCharsets.UTF_8)
    }

    @JvmStatic
    fun convertStringToUTF8(s: String): String {
        return String(s.toByteArray(StandardCharsets.UTF_8), StandardCharsets.ISO_8859_1)
    }
}